### ISSUE TITLE
Finetune on PASCAL VOC 2012 with detector COCO ckpt accuracy

### ISSUE THREAD

**Issue by Samjith888** [google/automl#479]
I'm finetuning by using coco ckpt.

1)Downloaded `efficientdet-d0` checkpoints and extracted
`wget https://storage.googleapis.com/cloud-tpu-checkpoints/efficientdet/coco/efficientdet-d0.tar.gz`

2) Following command used for training  
`python main.py --mode=train_and_eval  training_file_pattern=tfrecord/pascal*.tfrecord validation_file_pattern=tfrecord/pascal*.tfrecord  model_name=efficientdet-d0  model_dir=/tmp/efficientdet-d0-finetune  ckpt=efficientdet-d0  train_batch_size=16  --eval_batch_size=16 --eval_samples=12000  --num_examples_per_epoch=50000 --num_epochs=1000  --hparams=voc_config.yaml`

*******voc_config.yaml********
```
num_classes: 1
moving_average_decay: 0
learning_rate: 0.008
```
**Inference :**

I got AP50=0.388 after 8 epochs and Tested the model on a video , and also tested the pretrained coco efficientdet-d0 checkpoint on the same video.

**Observation :**
The inference by using "pretrained coco efficientdet-d0 checkpoint " is far better than the model which i got after 8 epoch finetunning. Atleast the inference accuracy should be same as the efficientdet-d0 coco ckpt, but its lesser than that.

**Tensorboad :**  

![lsnew](https://user-images.githubusercontent.com/39676803/83864438-5ab96100-a742-11ea-816c-14dac4a2ce63.png)

![gcls](https://user-images.githubusercontent.com/39676803/83864133-f72f3380-a741-11ea-9483-01b6a585fe1b.png)
![lsnew](https://user-images.githubusercontent.com/39676803/83864305-2fcf0d00-a742-11ea-9d02-6ed907346164.png)

Whats wrong here ? Any suggestion and helps will be appreciated

**Comment by mingxingtan** [comment]
You set num_classes=1, is it expected for PASCAL VOC 2012?

**Comment by Samjith888** [comment]
> You set num_classes=1, is it expected for PASCAL VOC 2012?

I have converted a custom dataset with one class into pascal voc format and converted into tf record followed by ur readme..

Yes i have only one class

**Comment by fsx950223** [comment]
```
num_classes: 2
label_id_mapping: {0: background, 1: classname}
learning_rate: 0.008
```

**Comment by Samjith888** [comment]
> num_classes: 2
> label_id_mapping: {0: background, 1: classname}
> learning_rate: 0.008

What value for moving_average_decay value ?

**Comment by MozhganPourKeshavarz** [comment]
I have the same issue,

after 13 epochs I got this result:
 ```
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
I0608 20:39:21.352850 139910415730496 evaluation.py:273] Inference Time : 39.78626s
I0608 20:39:21.353007 139910415730496 evaluation.py:276] Finished evaluation at 2020-06-08-20:39:21
I0608 20:39:21.353097 139910415730496 estimator.py:2066] Saving dict for global step 4635: AP = 0.0, AP50 = 0.0, AP75 = 0.0, APl = -1.0, APm = -1.0, APs = 0.0, ARl = -1.0, ARm = -1.0, ARmax1 = 0.0, ARmax10 = 0.0, ARmax100 = 0.0, ARs = 0.0, box_loss = 0.003286841, cls_loss = 0.010742487, global_step = 4635, loss = 0.26613998
I0608 20:39:21.353373 139910415730496 estimator.py:2127] Saving 'checkpoint_path' summary for global step 4635: efficientdet-d0-finetune/model.ckpt-4635
I0608 20:39:21.353670 139910415730496 error_handling.py:115] evaluation_loop marked as finished
I0608 20:39:21.353741 139910415730496 main.py:390] Evaluation results: {'AP': 0.0, 'AP50': 0.0, 'AP75': 0.0, 'APl': -1.0, 'APm': -1.0, 'APs': 0.0, 'ARl': -1.0, 'ARm': -1.0, 'ARmax1': 0.0, 'ARmax10': 0.0, 'ARmax100': 0.0, 'ARs': 0.0, 'box_loss': 0.003286841, 'cls_loss': 0.010742487, 'loss': 0.26613998, 'global_step': 4635}
```

I'm finetuning on a custom dataset with 2 class by using coco ckpt,
voc_config.yaml 
```
num_classes: 2
moving_average_decay: 0
learning_rate: 0.001
lr_warmup_init: 0.0001

```
system configuration : GTX 1080 ti
train command in terminal :
`python main.py --mode=train_and_eval --training_file_pattern=data/train.record --validation_file_pattern=data/val.record --model_name=efficientdet-d0 --model_dir=efficientdet-d0-finetune  --ckpt=efficientdet-d0  --train_batch_size=8 --eval_batch_size=8 --eval_samples=217 --num_examples_per_epoch=4122 --num_epochs=100 --hparams=voc_config.yaml --use_tpu=False
`

**Comment by dishen12** [comment]
have you solved this problem?@MozhganPourKeshavarz

**Comment by adesgautam** [comment]
Same, Ian getting loss 10 and it's constant throughout the training. It is not getting decreased.
